projects/igniteui-angular/src/lib/test-utils/ui-interactions.spec.ts

```diff
@@ -9,7 +9,21 @@ declare var Touch: {
     new(prop): Touch;
 };
 export class UIInteractions {
+    public static enterEvent = { key: 'Enter', stopPropagation: () => { }, preventDefault: () => { } };
+    public static endEvent = { key: 'End', stopPropagation: () => { }, preventDefault: () => { } };
+    public static homeEvent = { key: 'Home', stopPropagation: () => { }, preventDefault: () => { } };
+    public static spaceEvent = { key: 'Space', stopPropagation: () => { }, preventDefault: () => { }, stopImmediatePropagation: () => { } };
+    public static tabEvent = { key: 'Tab', stopPropagation: () => { }, preventDefault: () => { } };
     public static escapeEvent = { key: 'Escape', stopPropagation: () => { }, preventDefault: () => { } };
+    public static arrowDownEvent = { key: 'ArrowDown', stopPropagation: () => { }, preventDefault: () => { } };
+    public static altAndArrowDownEvent = { key: 'ArrowDown', altKey: true, stopPropagation: () => { }, preventDefault: () => { } };
+    public static arrowUpEvent = { key: 'ArrowUp', stopPropagation: () => { }, preventDefault: () => { } };
+    public static altAndArrowUpEvent = { key: 'ArrowUp', altKey: true, stopPropagation: () => { }, preventDefault: () => { } };
+    public static clickEvent = new MouseEvent('click');
+
+    public static getKeyboardEvent(eventType: string, keyboardEvent: KeyboardEventInit) {
```

Comment:
Can you change this method to:
 public static getKeyboardEvent(eventType: string, keyPressed: string, altKey = false, shift = false, ctrl = false) {
        const keyboardEvent = {
            key: keyPressed,
            altKey: altKey,
            shiftKey: shift,
            ctrlKey: ctrl,
            stopPropagation: () => { },
            stopImmediatePropagation: () => { },
            preventDefault: () => { }
        };
        return new KeyboardEvent(eventType, keyboardEvent);
    }
and remove enterEvent, endEvent, etc
and you can call it with:
combo.handleKeyUp(UIInteractions.getKeyboardEvent('keyup', 'ArrowDown')); for example